### ISSUE TITLE
how to mock authentication in integration tests

### ISSUE THREAD

**Issue by rburgst** [ByteInternet/drf-oidc-auth#50]
It would be useful to have some guidance on how to best come up with a valid token when used with `APIClient`

**Comment by rburgst** [comment]
I can see a quite laborious way to do it in https://github.com/ByteInternet/drf-oidc-auth/blob/master/tests/test_authentication.py but it would be still very useful to have a condensed version of this mentioned in the readme.

**Comment by kerrermanisNL** [comment]
Hi rburgst, good suggestion, I will have a look if I can add something on Friday.

**Comment by kerrermanisNL** [comment]
@rburgst in your case, would it be enough for you to do something like this?
```
class MyTestCase(AuthenticationTestCase):
    def test_my_integration_test(self):
        self.responder.set_response('https://mywebsite.com/userinfo', 'sub': 'my_username)

        resp = self.client.get('/myurl/', HTTP_AUTHORIZATION=auth)

        // Test something
```

If so, I might move the `AuthenticationTestCase` etc. inside the module so people can use it for their tests. But it's a bit hard for me to see if this would be enough for you.

**Comment by rburgst** [comment]
I would prefer to have this as a mixin as we already have a couple of test base classes that I dont want to touch, if possible.

**Comment by kerrermanisNL** [comment]
Sure that's possible. As long as you can confirm it will help you out :)

**Comment by rburgst** [comment]
FYI: this is what I am currently using which is pretty much copied from your authentication test case

´´´python

class BaseAuthenticationTest(BaseAPITest):
    username_granted = 'APIUser'

    def patch(self, thing_to_mock, **kwargs):
        patcher = patch(thing_to_mock, **kwargs)
        patched = patcher.start()
        self.addCleanup(patcher.stop)
        return patched

    def setUp(self):
        self.responder = FakeRequests()
        self.responder.set_response(
            "https://example.com/.well-known/openid-configuration",
            {"jwks_uri": "http://example.com/jwks",
             "issuer": "http://example.com",
             "userinfo_endpoint": "http://example.com/userinfo"})
        self.mock_get = self.patch('requests.get')
        self.mock_get.side_effect = self.responder.get
        keys = KeySet(keys=[key])
        self.patch(
            'oidc_auth.authentication.request',
            return_value=Mock(
                status_code=200,
                json=keys.as_json
            )
        )

    def mock_authentication(
        self, ...
    ):
        self.responder.set_response(
            'http://example.com/userinfo',
            {
                "language": "de",...
                })
```

**Comment by kerrermanisNL** [comment]
See https://github.com/ByteInternet/drf-oidc-auth/pull/53
Will this work for you?

**Comment by rburgst** [comment]
This is quite good. I would however need to configure the created username (right now “henk” is hardcoded)

**Comment by kerrermanisNL** [comment]
Fixed in #53 will tag a release later on
